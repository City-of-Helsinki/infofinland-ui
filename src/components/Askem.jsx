import cls from 'classnames'
import { useAskem } from '@/hooks/useAskem'
import { useRouter } from 'next/router'

export const Askem = ({locale}) => {
  useAskem(locale);
  return (
    <section className="feedback">
      <div
        className={cls(
          'askem px-2 lg:px-0 lg:mx-12 xl:mx-28 2xl:mx-48  3xl:ms-64  3xl:max-w-4xl',
          'h-20'
        )}
      >
        askem askem {locale}
      </div>
    </section>
  )
}
