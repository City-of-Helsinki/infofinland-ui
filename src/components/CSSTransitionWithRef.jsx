import { cloneElement, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';

function CSSTransitionWithRef({ children, ...restProps }) {
  const ref = useRef(null);

  const setRef = (value) => {
    ref.current = value;
  };

  return (
    <CSSTransition nodeRef={ref} {...restProps}>
      {cloneElement(children, { ref: setRef })}
    </CSSTransition>
  );
}

export { CSSTransitionWithRef };