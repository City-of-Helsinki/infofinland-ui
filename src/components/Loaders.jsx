import cls from 'classnames'

export const DotsLoader = ({ color = 'blue' }) => (
  <div className={cls('ifu-loader__feedback--ellipsis m-auto block', color)}>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
  </div>
)

export const RingsLoader = ({ color = '' }) => (
  <div className="flex absolute justify-items-center items-center w-full h-full">
    <div
      className={cls('scale-75 lg:scale-100 ifu-loader__image--ripple', color)}
    >
      <div></div>
      <div></div>
    </div>
  </div>
)
