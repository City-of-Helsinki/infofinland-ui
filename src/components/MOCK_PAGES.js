//Demo data
export const PAGES = [
  { text: 'Home', url: '/' },
  {
    text: 'Moving to Finland',
    url: '/theme1',

    pages: [
      { text: 'Health frontpage', url: '/theme1/1' },
      { text: ' Emergencies', url: '/theme1/2' },
      {
        text: ' Health services in Finland',
        url: '/theme1/3',

        pages: [
          { text: 'Health frontpage', url: '/theme1/3/1' },
          { text: ' Emergencies', url: '/theme1/3/2' },
          { text: ' Health services in Finland', url: '/theme1/3/3' },
          { text: ' Support during illness', url: '/theme1/3/4' },
        ],
      },

      { text: 'Medication', url: '/theme1/4' },
      { text: ' Support during illness', url: '/theme1/5' },
      { text: 'Rehabilitation', url: '/theme1/6' },
      { text: ' Disabled persons', url: '/theme1/7' },
    ],
  },
  {
    text: ' Education',
    url: '/theme2',
    pages: [
      { text: 'Childrenâ€™s health', url: '/theme2/1' },
      { text: 'Dental care', url: '/theme2/2' },
      { text: 'Mental health', url: '/theme2/3' },
      { text: 'Sexual health and contraception', url: '/theme2/4' },
      { text: 'Abortion', url: '/theme2/5 ' },
    ],
  },
  // {
  //   text: 'Settling in Finland',
  //   url: '/theme009',
  //   pages: SUBMENU,
  // },
  // {
  //   text: 'Work and enterprise',
  //   url: '/theme3',
  //   pages: [
  //     { text: 'Health frontpage', url: '/asdad' },
  //     { text: ' Emergencies', url: '/asdasd' },
  //     { text: ' Health services in Finland', url: '/asdasdsa' },
  //   ],
  // },

  // {
  //   text: 'Finnish and Swedish',
  //   url: '/theme4',
  //   pages: [
  //     {
  //       text: 'Health frontpage',
  //       url: '/asdadsa2',
  //       pages: [
  //         { text: 'Health frontpage', url: '/324234' },
  //         { text: ' Emergencies', url: '/2342g2' },
  //         { text: ' Health services in Finland', url: '/32432' },
  //         { text: ' Support during illness', url: '/343g43' },
  //         { text: 'Rehabilitation', url: '/365e7n' },
  //         { text: ' Disabled persons', url: '/5b4326856' },
  //       ],
  //     },
  //     { text: ' Emergencies', url: '/63w45b' },
  //     { text: ' Health services in Finland', url: '/34b36' },
  //     { text: ' Support during illness', url: '/g6' },
  //     { text: 'Rehabilitation', url: '/25434h45' },
  //     { text: ' Disabled persons', url: '/jugi23' },
  //   ],
  // },
  // {
  //   text: 'Housing',
  //   url: '/theme5',
  //   pages: [
  //     { text: 'Pregnancy and childbirth', url: '/sebw35' },
  //     { text: 'Medication', url: '/3b5435' },
  //     { text: ' Support during illness', url: '/34554n5' },
  //     { text: 'Rehabilitation', url: '/ser264' },
  //     { text: ' Disabled persons', url: '/sfdb243' },
  //   ],
  // },

  // { text: ' Health', url: '/theme123', pages: SUBMENU },
  // { text: ' Family', url: '/theme45674745' },
  // { text: ' Leisure', url: '/theme', pages: SUBMENU },
  // { text: ' Information', url: '/theme22362532' },
  // { text: ' About Finland', url: '/sar42342', pages: SUBMENU },
]
