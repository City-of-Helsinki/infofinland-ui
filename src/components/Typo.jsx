import cls from 'classnames'

export const longTextClass = (text, { size, classes: [short, long] }) =>
  cls({
    [short]: text.length < size,
    [long]: text.length >= size,
  })

export const H1 = ({ children, text, className }) => (
  <h1
    className={cls(
      'text-bodytext-color font-sans text-h1 md:text-h1xl',
      className
    )}
  >
    {text}
    {children}
  </h1>
)

export const H2 = ({ children, className }) => (
  <h2
    className={cls(
      'text-bodytext-color font-sans text-h2 md:text-h2xl',
      className
    )}
  >
    {children}
  </h2>
)

export const H3 = ({ children, className }) => (
  <h3
    className={cls(
      'text-bodytext-color font-sans text-h3 md:text-h3xl',
      className
    )}
  >
    {children}
  </h3>
)

export const H4 = ({ children, className, bold }) => (
  <h4
    className={cls(
      'text-bodytext-color font-sans text-h4 md:text-h4xl',
      className,
      {
        'font-bold': bold,
      }
    )}
  >
    {children}
  </h4>
)

export const H5 = ({ children, className, bold }) => (
  <h4
    className={cls(
      'text-bodytext-color font-sans text-h5 md:text-h5xl',
      className,
      {
        'font-bold': bold,
      }
    )}
  >
    {children}
  </h4>
)

export const H6 = ({ children, className, bold }) => (
  <h4
    className={cls('text-bodytext-color font-sans text-h6', className, {
      'font-bold': bold,
    })}
  >
    {children}
  </h4>
)

export const HR = ({ className }) => <hr className={cls('my-8', className)} />
