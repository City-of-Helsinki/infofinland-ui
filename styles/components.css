@import './components.messagecard.css';
@import './components.mainmenu.css';
@import './components.drawer.css';
@import './components.modal.css';
@import './components.topmenu.css';
@import './components.hero.css';
@import './components.loaders.css';
@import './components.feedback.css';
@import './components.article.css';
@import './components.footer.css';
@import './components.local-info.css';
@import './components.search.css';
@import './components.cookies.css';
@import './components.accordion.css';

@layer components {
  .ifu-layout__body {
    min-height: calc(100vh - 80px);
    @apply flex flex-col justify-between  flex-1 xl:px-0 border-black-op1 border-s-2 overflow-hidden;
  }

  .ifu-text-link {
    @apply text-link hover:text-link-hover hover:underline;
  }

  .ifu-button {
    @apply text-black bg-white shadow-button hover:shadow-button-hover border-2 border-bodytext-color rounded px-4 py-2 leading-6  font-sans font-bold inline-block;
  }

  .ifu-button:focus-visible {
    outline-position: outside;
    outline-style: solid;
    outline-color: rgba(0, 0, 50, 0.3);
    outline-width: 3px;
  }

  .ifu-breadcrumb__icon--arrow {
    @apply transform scale-150;
    --tw-translate-y: -0.04rem;
  }

  .ifu-breadcrumb__icon--home {
    @apply transform scale-125;
    --tw-translate-y: -0.22rem;
  }
}
