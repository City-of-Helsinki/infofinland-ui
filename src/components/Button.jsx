import cls from 'classnames'
import { IconCross } from '@/components/Icons'

import { useTranslation } from 'next-i18next';

const Button = ({ children, className, ...buttonProps }) => (
  <button className={cls('ifu-button', className)} {...buttonProps}>
    {children}
  </button>
)

export default Button

export const LinkButton = ({ children, className, ...buttonProps }) => (
  <button
    className={cls(
      'ifu-text-link  px-4 py-2 leading-6  font-sans font-bold',
      className
    )}
    {...buttonProps}
  >
    {children}
  </button>
)

export const CloseButton = ({ close, className }) => {
  const { t } = useTranslation('common')

  return (
    <button
      className={cls('flex items-center mt-4 me-3', className)}
      onClick={close}
    >
      <span className="inline-block px-1 text-small">{t('buttons.close')}</span>
      <IconCross className="inline-block text-black transform -translate-y-0.5 fill-current" />
    </button>
  )
}
