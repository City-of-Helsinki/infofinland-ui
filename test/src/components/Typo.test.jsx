/**
 * @jest-environment jsdom
 */

import React from 'react'
// Using render and screen from test-utils.js instead of
// @testing-library/react
// import { render } from "@/test/test-utils";
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import { H1, H2, H3, H4, H5, H6, HR } from '@/components/Typo'

describe('Typography', () => {
  it('should render the h1', () => {
    const { asFragment } = render(<H1>Heading 1</H1>)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the h2', () => {
    const { asFragment } = render(<H2>Heading 2</H2>)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the h3', () => {
    const { asFragment } = render(<H3>Heading 3</H3>)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the h4', () => {
    const { asFragment } = render(<H4>Heading 4</H4>)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the h5', () => {
    const { asFragment } = render(<H5>Heading 5</H5>)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the h6', () => {
    const { asFragment } = render(<H6>Heading 6</H6>)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render the hr', () => {
    const { asFragment } = render(<HR />)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
