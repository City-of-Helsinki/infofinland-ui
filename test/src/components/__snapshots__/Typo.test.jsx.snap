// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render the h1 1`] = `
<DocumentFragment>
  <h1
    class="text-bodytext-color font-sans text-h1 md:text-h1xl"
  >
    Heading 1
  </h1>
</DocumentFragment>
`;

exports[`Typography should render the h2 1`] = `
<DocumentFragment>
  <h2
    class="text-bodytext-color font-sans text-h2 md:text-h2xl"
  >
    Heading 2
  </h2>
</DocumentFragment>
`;

exports[`Typography should render the h3 1`] = `
<DocumentFragment>
  <h3
    class="text-bodytext-color font-sans text-h3 tmd:text-h3xl"
  >
    Heading 3
  </h3>
</DocumentFragment>
`;

exports[`Typography should render the h4 1`] = `
<DocumentFragment>
  <h4
    class="text-bodytext-color font-sans text-h4 md:text-h4xl"
  >
    Heading 4
  </h4>
</DocumentFragment>
`;

exports[`Typography should render the h5 1`] = `
<DocumentFragment>
  <h4
    class="text-bodytext-color font-sans text-h5 md:text-h5xl"
  >
    Heading 5
  </h4>
</DocumentFragment>
`;

exports[`Typography should render the h6 1`] = `
<DocumentFragment>
  <h4
    class="text-bodytext-color font-sans text-h6"
  >
    Heading 6
  </h4>
</DocumentFragment>
`;

exports[`Typography should render the hr 1`] = `
<DocumentFragment>
  <hr
    class="my-8"
  />
</DocumentFragment>
`;
